src/core/rules.js

```diff
@@ -562,41 +676,54 @@ Rules.typeFeaturesMatch = function typeFeaturesMatch(tf1, tf2) {
 };
 
 class AndRules extends Rules {
-    constructor(rules) {
-        super();
+    constructor(rules, excludeOn) {
+        super(excludeOn);
         this.add(rules);
     }
 
-    async evaluate(subject, type, feature, instance, featureMap, instanceManager) {
+    async evaluate(subject, type, feature, instance, featureMap, instanceManager, parentRule) {
+        if (this.shouldExcludeOn(type, feature)) {
+            return parentRule === null ? false : null;
+        }
+
+        let andFulfilled = true;
+
         for(const rule of this.rules) {
-            if (!(await rule.evaluate(subject, type, feature, instance, featureMap, instanceManager))) {
-                return false;
```

Comment:
to be consistent: andFulFilled = false + else if?

Reply:
The evaluation can return null (i.e. rule is ignored by excludeOn) or true. If it evaluates to false, the AND rule evaluates to false.

Reply:
I meant more that there are a mix up between to patterns --> default return and then return at the end vs. direct return. Logically this is would be the same but it has some 'taste' of inconsistency.

You could set andFulfilled = false and call a break instead of a direct return. (I was wrong in the first place, I haven't recognized that we are in a loop here)

Reply:
It depends whether you would like to continue the iteration or stop it. In both rule evaluation functions, we need to go through the whole loop to be able to fill out all requested features in the given featureMap.

Reply:
The newest change looks good 👍